### PR TITLE
#2940 Reason modal to quick

### LINKED ISSUE #2940: Reason modal pops up to quick on a supplier requisition
## Describe the bug

Can't enter more than 1 digit on supplier requisition quantity column

### To reproduce

1. Ivory coast data file
1. Go to 'supplier requisition'
2. Tap on 'requested quantity'
3. Try to put the requested quantity with >1 digits
4. You will be prompted to choose the reason

### Proposed Solution

Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info

- App version: 5.0.1
- Tablet model: Samsung 8'
- OS version: Android 9

### Additional context

see video
https://www.dropbox.com/s/ni35fcwx8im8wdw/supplierrequisition-1.mp4?dl=0

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2940 

## Change summary

- Makes Requisition reasons consistent with stocktakes: No enforcement
- Added removal of reasons when there is no variance to the standard stocktake counted quantity editing action, which means you can remove a lot of code. No difference in actual behaviour for stocktakes

## Testing

- [ ] Can add a reason to a requisition line when the line has a variance between requested and suggested
- [ ] When there is no variance the reason cell is disabled
- [ ] When editing a line from having a variance to not having a variance, the reason is removed.

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/DataTable/DataTableRow.js [hunk 1]
```diff
@@ -324,18 +319,19 @@ const DataTableRow = React.memo(
...
-              const { isVaccine = false } = rowData ?? {};
+              const { isVaccine = false, hasVariance } = rowData ?? {};
 
               const disabledConditions = {
                 [COLUMN_KEYS.CURRENT_VVM_STATUS]: !isVaccine,
+                [COLUMN_KEYS.REASON_TITLE]: !hasVariance,
               };
 
-              const disabledVVMStatus = disabledConditions[columnKey];
+              const extraDisabledCondition = disabledConditions[columnKey];
```
Comment (wlthomson) [inline src/widgets/DataTable/DataTableRow.js:329]: blewmind

...maybe something for a tidy up later?

**Review by wlthomson — APPROVED** [review]
lgtm

File: src/widgets/DataTable/DataTableRow.js [hunk 2]
```diff
@@ -324,18 +319,19 @@ const DataTableRow = React.memo(
             }
 
             case COLUMN_TYPES.DROP_DOWN: {
-              const { isVaccine = false } = rowData ?? {};
+              const { isVaccine = false, hasVariance } = rowData ?? {};
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/widgets/DataTable/DataTableRow.js]: f5ae87659110ccabf7fe23d8a002649f9c6ab3f4

File: src/pages/dataTableUtilities/actions/cellActions.js [hunk 3]
```diff
@@ -523,17 +462,15 @@ export const CellActionsLookup = {
...
-  editStocktakeBatchCountedQuantityWithReason,
   removeReason,
   applyReason,
   editBatchName,
   editIndicatorValue,
   editStocktakeBatchName,
   editSellPrice,
   editBatchSupplier,
-  editRequisitionItemRequiredQuantityWithReason,
+  // editRequisitionItemRequiredQuantityWithReason,
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/dataTableUtilities/actions/cellActions.js]: 5b1dac54eb9965a289aedc3be8ecc065078bd373

File: src/pages/dataTableUtilities/actions/cellActions.js [hunk 4]
```diff
@@ -380,50 +377,19 @@ export const editRequiredQuantity = (value, rowKey, objectType, route) => (dispa
       objectToEdit.requiredQuantity = parsePositiveInteger(value);
       UIDatabase.save(objectType, objectToEdit);
     });
+
+    if (!objectToEdit?.hasVariance) dispatch(removeReason(rowKey, route));
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/dataTableUtilities/actions/cellActions.js]: 0bb2072ec4ca5cb387113ec9529a97a7189cf817

Changed a bit and also fixed the logic for stocktake batches which weren't removing the reason when going from positive -> negative and vice versa

**Review by wlthomson — APPROVED** [review]

**Pull request merged**